portfolio/src/main/java/com/google/sps/servlets/UserSubmittedCommentsServlet.java

```diff
@@ -32,40 +32,59 @@
+@WebServlet("/submitted-comments")
+public class UserSubmittedCommentsServlet extends HttpServlet {
   private static final String NAME = "name";
   private static final String PHONE = "phone";
   private static final String LOCATION = "location";
   private static final String DESCRIPTION = "description";
   private static final String TABLE_NAME = "Pins";
   private static final String TIME_STAMP = "timestamp";
+  private static final String ID = "id";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    Query query = new Query(TABLE_NAME);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<UserComment> userComments = new ArrayList<UserComment>();
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty(NAME);
+      String phone = (String) entity.getProperty(PHONE);
+      String location = (String) entity.getProperty(LOCATION);
+      String description = (String) entity.getProperty(DESCRIPTION);
+      long timeStamp = (long) entity.getProperty(TIME_STAMP);
+      long id = entity.getKey().getId();
+
+      UserComment userComment = new UserComment(name, phone, location, description, timeStamp, id);
+      userComments.add(userComment);
+    }
+
+    response.setContentType("application/json");
+    response.getWriter().println(new Gson().toJson(userComments));
+  }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    System.out.println("IN DO POST----------------------------------");
     UserService userService = UserServiceFactory.getUserService();
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     String name = request.getParameter(NAME);
     String phone = request.getParameter(PHONE);
     String location = request.getParameter(LOCATION);
     String description = request.getParameter(DESCRIPTION);
     long timeStamp = System.currentTimeMillis();
-    System.out.println(name);
-    System.out.println(phone);
-    System.out.println(location);
-    System.out.println(description);
 
     Entity pinEntity = new Entity(TABLE_NAME);
-    pinEntity.setProperty("Name", name);
-    pinEntity.setProperty("Phone", phone);
-    pinEntity.setProperty("Address", location);
-    pinEntity.setProperty("Description", description);
-    pinEntity.setProperty("TimeStamp", timeStamp);
+    pinEntity.setProperty(NAME, name);
+    pinEntity.setProperty(PHONE, phone);
+    pinEntity.setProperty(LOCATION, location);
+    pinEntity.setProperty(DESCRIPTION, description);
+    pinEntity.setProperty(TIME_STAMP, timeStamp);
     datastore.put(pinEntity);
 
     response.setContentType("text/html");
-    response.getWriter().println("TEST");
     response.sendRedirect("/statistics.html");
```

Comment:
After user post a location on map, why redirecting to stats page?

Reply:
Once the form is submitted, it would redirect the user to the /submitted-comments endpoint that is empty. In order to prevent this redirection, I redirect the user back to /statistics.html.

Reply:
re: the action="/submitted-comments" value in the form tag

Reply:
Then we should figure out why posting a new location will redirect user instead of having this workaround.

Reply:
You'd have to submit the form using ajax instead to prevent a redirect. That could be a whole can of worms so I'm okay with a TODO here instead.

If you are curious, the answer here is an example of how to do that (ignore the php part):
https://stackoverflow.com/questions/5004233/jquery-ajax-post-example-with-php